Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
```

Comment:
This is however a completely valid concern in the `except` case. If the exception does not fit the standard colon delimited format then we can't make any assumptions about its encoding. Which is probably why it was using `repr(errpipe_data)` before.